src/main/java/com/github/jrubygradle/jem/JemsClassLoader.java

```diff
@@ -0,0 +1,50 @@
+package com.github.jrubygradle.jem;
+
+import com.github.jrubygradle.jem.internal.GemInstaller;
+import com.github.jrubygradle.jem.GemInstaller.DuplicateBehavior;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.nio.file.Files;
+import java.util.Collections;
+
+/**
+ * Created by cmeier on 8/15/15.
+ */
+public class JemsClassLoader extends URLClassLoader {
+
+    private final File jemsDir;
+
+    public static JemsClassLoader create() throws IOException {
+        return new JemsClassLoader(Files.createTempDirectory("jems").toFile());
+    }
+
+    public JemsClassLoader(File jemsDir) throws IOException {
+        super(new URL[] { jemsDir.toURI().toURL() });
+        this.jemsDir = jemsDir;
+    }
+
+    public Gem addJem(String jem) throws IOException {
+        return addJem(new File(jem));
+    }
+    public Gem addJem(File jem) throws IOException {
```

Comment:
yes, we can build a classloader which can do this. BUT it needs to register its own url.protocol.handler by setting a system property (or appending to the existing value) and this can fail. so we always need a fallback in such a case. I am not sure if I really want to have all the gems puts permanently into memory - they are just needed until all ruby scripts are loaded. probably some soft references (or weak ones) can help here.

JRuby needs a "string" for the $LOAD_PATH that is the reason for the url protocol handler.